apps/settings/components/avatar_uploader/view.coffee

```diff
@@ -0,0 +1,83 @@
+Promise = require 'bluebird-q'
+Backbone = require 'backbone'
+{ API_URL } = require('sharify').data
+guid = require './lib/guid.coffee'
+fileTypeRegex = require './lib/file_type_regex.coffee'
+template = -> require('./index.jade') arguments...
+
+module.exports = class AvatarUploaderView extends Backbone.View
+  class: 'avatar-uploader'
+
+  events:
+    'click .js-submit': 'proxyClick'
+
+  initialize: ->
+    @listenTo @model, 'change:avatar_image', @render
+
+  proxyClick: (e) ->
+    e.preventDefault()
+    @$('input:file').trigger('click')
+
+  save: (url) ->
+    $.ajax
+      type: 'PUT'
+      # TODO: allow this to be updated through normal user POST
+      url: "#{API_URL}/accounts/avatar"
+      data: url: url
+
+    poll = () =>
+      @model.fetch().then =>
+        unless @model.hasChanged 'avatar_image'
+          setTimeout poll, 1000
+        else
+          @els.submit.text 'Avatar updated'
+
+    poll()
```

Comment:
Nice, fwiw, we still send a pusher update for this. channel: "user_#{id}", action: "user-#{updated}". Poll-ing might be easier though.